gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/rest/HttpJsonServiceStubClassComposer.java

```diff
@@ -1103,6 +1150,73 @@ protected List<Expr> createOperationsStubInitExpr(
   }
 
+  /* Build an Expr that creates an HttpRule. Creates a builder and adds the http verb, custom path, and any additional bindings. `additional_bindings` can only be nested one layer deep, so we only check once */
+  private Expr createHttpRuleExpr(HttpRule httpRule, boolean checkAdditionalBindings) {
+    Expr httpRuleBuilderExpr =
+        MethodInvocationExpr.builder()
+            .setStaticReferenceType(FIXED_REST_TYPESTORE.get(HttpRule.class.getSimpleName()))
+            .setMethodName("newBuilder")
+            .build();
+
+    httpRuleBuilderExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(httpRuleBuilderExpr)
+            // toLowerCase as the PatternCase result is all uppercase
+            .setMethodName(setMethodFormat(httpRule.getPatternCase().toString().toLowerCase()))
+            .setArguments(
+                ValueExpr.withValue(
+                    StringObjectValue.withValue(getOperationsURIValueFromHttpRule(httpRule))))
+            .setReturnType(FIXED_REST_TYPESTORE.get(HttpRule.class.getSimpleName()))
+            .build();
+
+    if (checkAdditionalBindings) {
+      for (HttpRule additionalBindings : httpRule.getAdditionalBindingsList()) {
+        httpRuleBuilderExpr =
+            MethodInvocationExpr.builder()
+                .setExprReferenceExpr(httpRuleBuilderExpr)
+                .setMethodName("addAdditionalBindings")
+                .setArguments(Arrays.asList(createHttpRuleExpr(additionalBindings, false)))
+                .build();
+      }
+    }
+
+    httpRuleBuilderExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(httpRuleBuilderExpr)
+            .setMethodName("build")
+            .setReturnType(FIXED_REST_TYPESTORE.get(HttpRule.class.getSimpleName()))
+            .build();
+    return httpRuleBuilderExpr;
+  }
+
+  /* Parses the Service Yaml file's for custom HttpRules. Filter the HttpRules for ones that match Operations */
+  private Map<String, HttpRule> parseOperationsCustomHttpRules(GapicContext context) {
+    Predicate<HttpRule> predicate = x -> x.getSelector().contains(LRO_NAME_PREFIX);
+    com.google.api.Service service = context.serviceYamlProto();
+    if (service == null || service.getHttp() == null) {
+      return ImmutableMap.of();
+    }
+    return service.getHttp().getRulesList().stream()
+        .filter(predicate)
+        .collect(Collectors.toMap(HttpRule::getSelector, x -> x));
+  }
+
+  /* This is meant to be used for the OperationsClient Mixin OperationsClient's RPCs are mapped to GET/POST/DELETE and this function only expects those HttpVerbs to be used */
+  private String getOperationsURIValueFromHttpRule(HttpRule httpRule) {
+    switch (httpRule.getPatternCase().getNumber()) {
+      case 2:
+        return httpRule.getGet();
+      case 4:
+        return httpRule.getPost();
+      case 5:
+        return httpRule.getDelete();
+      default:
+        throw new IllegalArgumentException(
```

Comment:
I would probably write some true unit tests for this method and the method above, as the input and output are very easy to mock and verify, but I know they are mostly already covered in the golden unit tests, so as long as we covered all the cases, it's fine to me. 
However, this branch couldn't be covered by golden tests, so we should add a unit test for it(by changing the scope of this method to default).

Reply:
> However, this branch couldn't be covered by golden tests, so we should add a unit test for it

I'm a bit lost of this part. The Echo goldens for rest should have it right?
https://github.com/googleapis/gapic-generator-java/blob/88487e38f86494e2ba18d4bf7c7f23c6671aeb24/gapic-generator-java/src/test/java/com/google/api/generator/gapic/composer/rest/goldens/HttpJsonEchoStub.golden#L360-L405